### PR TITLE
Restructure model build

### PR DESCRIPTION
**Pull request by AAmedeo**
## TLDR

Currently the ModelBuild class packs too many functionalities which makes it hard to work on the code.
These changes goal is to split the ModelBuild in multiple parts each with a specific function. 

## The new structure 
The 4 main functions that ModelBuild contains right now are:
1. Define all the variables for the model
2. Define all the constraints for the model
3. Define the objective functions for the model
4. Create and run the cvxpy problem instance and collect the results

With this change:
1. The functionality to create the variables for the model will be in a class called ModelVariables
2. The functionality to create the model constraints will be in a separate file for each constraint.
3. The definition of the objective functions and the functionality to create and run the cvxpy model will stay in ModelBuild 

Additionally I will move to the constraint definition files the definitions of constraint specific model parameters that are relevant only to a specific constraint. For example the line_mintotcap and line_maxtotcap parameters are only ever used by the LineTotalCapacity constraints, therefore they will be defined in the same file as that constraint. 

## The commits
[**Create ModelVariables**](https://github.com/AAmedeo/Hypatia/commit/acfa8af9df40e3efd88ff705a1fcea9b60ff4179)
Moves the mode variables definition from ModelBuild to ModelVariables

[**Move each constraints to separate file**](https://github.com/AAmedeo/Hypatia/commit/f9daa5e04b63f3e0e1883d43dab1a5793b1e886a)
Moves constraints definition from ModelBuild to a separate file for each constraint

[**Move constraint specific parameters definition in constraint files**](https://github.com/AAmedeo/Hypatia/commit/444f65b9221ca7c7fc8f22a25ca6a043c898a138)
Moves constraint specific parameters definition from ModelSettings to the appropriate constraint file. 

## Tests
This changes were expensively tested manually while working on the Italy2020 project (all this code was used to generate the results for that project). 

Ideally we would test this in a more formal way by creating a test suite that cover the model creation and optimization steps (i.e. tests that create model in a way that a specific feature will be tested, run it, and check that the mode results are correct). 

However, discussing with Francesco offline we decided that we are ok with merging this to dev before the full tests suite is created (especially given how much manual testing was done to this code during the Italy2020 project). This has the benefit of unblocking other people that need to start working on top of this code without waiting for all the unit tests to be ready. 

Before this is merged to the main branch the plan is to create a full suite of unit tests that will check this code functionality. 

Here is an example of the tests we are going to add before merging these changes to main: https://github.com/AAmedeo/Hypatia/commit/b9f41d083527d4f37f345d967f3d988146db1f95

### CODE REVIEW

**Comment by Negnamazi** [comment]
@AAmedeo 
Thank you for the changes.
I've done a general review so far! I will merge it to dev until a proper test suite will be created.
